Source/Controllers/Preferences/Panes/SPFilePreferencePane.m

```diff
@@ -501,7 +501,15 @@ - (void)chooseFileWithOptions:(PanelOptions*)options
                 // use url from the block, not self->_currentFilePanel.URL
                 // From Apple docs: The NSOpenPanel subclass sets this property to nil
                 // when the selection contains multiple items.
-                if([SecureBookmarkManager.sharedInstance addBookmarkForUrl:url options:(NSURLBookmarkCreationWithSecurityScope|NSURLBookmarkCreationSecurityScopeAllowOnlyReadAccess) isForStaleBookmark:options.isForStaleBookmark] == YES){
+
+                NSURLBookmarkCreationOptions bookmarkCreationOptions = (NSURLBookmarkCreationWithSecurityScope|NSURLBookmarkCreationSecurityScopeAllowOnlyReadAccess);
+
+                // try to make known_hosts files read-write
+                if([url.absoluteString contains:SPSSHKnownHostsFile] == YES){
```

Comment:
What if the file isn't named `known_hosts`? Technically they could name the file anything they want. Perhaps instead we pass an arg to this method that indicates if the bookmark should be write as well as read?

Reply:
good idea...

Reply:
tricky. It's a different screen. Knows nothing about what sort of file it is.